tools/release.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+// release.go creates a new version tag and pushes it to GitHub. It has two
+// main modes of operation:
+// * default: fetch origin/main, tag that commit with Major.Minor.Patch version
+//   of 0.YYYYMMDD.0, and push the tag.
+// * hotfix: fetch the named release tag, create a new branch starting from
+//   that commit, cherry-pick the named commit(s) onto that branch, tag the
+//   new branch head with a Major.Minor.Patch that matches the Major.Minor of
+//   named release tag and a Patch that is one greater, and push that tag.
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"golang.org/x/mod/semver"
+)
+
+func runOrDie(cmd *exec.Cmd) string {
+	fmt.Println(cmd.String())
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		fmt.Print(string(out))
```

Comment:
Not sure if you are intentionally `fmt.Print()` instead of `fmt.Println()` here.

Reply:
It was intentional (because the thing being output ends with a newline anyway), but in hindsight I think adding an extra empty newline for a bit of vertical spacing is probably better.